### ISSUE TITLE
Installation error with Python 3.8 and Visual Studio

### ISSUE THREAD

**Issue by mommenik** [lucasrodes/whatstk#102]
Hi, does know anybody what is not working here? Cheers

`PS C:\Users\Notebook\Desktop\test_python3> pip install whatstk
>>
...
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 2117: character maps to <undefined>
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

**Comment by lucasrodes** [comment]
Hi @mommenik,
First of all thanks for using `whatstk`. I am sorry I could not reach out before. 

Could you solve the problem? I did not test it in Windows, could you elaborate on which setup you are using?

**Comment by saviols** [comment]
Hello @lucasrodes . The installation is not working for me. Follows below the error message:

C:\Users\conta>python -m pip install --upgrade whatstk
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: whatstk in c:\users\conta\appdata\roaming\python\python37\site-packages (0.1.11)
Collecting whatstk
  Using cached whatstk-0.4.0.tar.gz (36 kB)
    ERROR: Command errored out with exit status 1:
     command: 'C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\conta\\AppData\\Local\\Temp\\pip-install-yrta14yd\\whatstk_a1c2f99b922e4aa9b343af87f0937c2f\\setup.py'"'"'; __file__='"'"'C:\\Users\\conta\\AppData\\Local\\Temp\\pip-install-yrta14yd\\whatstk_a1c2f99b922e4aa9b343af87f0937c2f\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\conta\AppData\Local\Temp\pip-pip-egg-info-ug2eoh1v'
         cwd: C:\Users\conta\AppData\Local\Temp\pip-install-yrta14yd\whatstk_a1c2f99b922e4aa9b343af87f0937c2f\
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\conta\AppData\Local\Temp\pip-install-yrta14yd\whatstk_a1c2f99b922e4aa9b343af87f0937c2f\setup.py", line 10, in <module>
        long_description = f.read()
      File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 2384: character maps to <undefined>

**Comment by lucasrodes** [comment]
Hi @saviols, I am afraid that at the moment this library can only be installed in Linux and Mac OS. 

However, I would like it to be also installable on Windows, so I'll keep this in mind, but currently don't have the time as my expertise in packaging python projects for Windows is limited. Contributions welcome 😄 

I'd like to address this in the future, but for the time being, I'd recommend, if possible, using Linux/Mac or docker to recreate a Linux environment.

**Comment by lucasrodes** [comment]
Hi @saviols,
I am currently working in making the library available in Windows.

Could you try installing it using the following command:

```
pip install git+https://github.com/lucasrodes/whatstk.git@feature/wheels
```

**Comment by saviols** [comment]
Hay Lucas,
It´s working. Now I can import and read the txt file. Well done.

However, there are some commands in your article "Exploring WhatsApp data"
that are not working.

Please let me know your email address so I can share the notebook with you.



Em seg., 24 de mai. de 2021 às 20:22, Lucas Rodés-Guirao <
***@***.***> escreveu:

**Comment by lucasrodes** [comment]
HI @saviols, 
Thanks for your response. 

Not sure if is gonna solve anything, but try the following instead:

```
pip install git+https://github.com/lucasrodes/whatstk.git@develop
```

To reach me out, can you do that via Twitter [@lucasrodesg](twitter.com/lucasrodesg). Ping me there so I can share my mail privately.

Kind regards

**Comment by lucasrodes** [comment]
Hi @saviols,
Could you share more details on which errors you are getting? I would appreciate that very much so I can make the library Windows compatible.

Make sure to have the library installed from the develop branch:

```
pip install git+https://github.com/lucasrodes/whatstk.git@develop
```


Thank you very much for your time. Also find my contact details [here](https://lcsrg.me/pages/contact.html)

**Comment by saviols** [comment]
# Day with the most number of messages sent
chat.df.groupby(chat.df.index.date).agg(num_messages=('message', 'count')).sort_values(by='num_messages', ascending=False)

Error Message:
AttributeError                            Traceback (most recent call last)
<ipython-input-7-7f3406c4c6bc> in <module>()
      1 # Day with the most number of messages sent
----> 2 chat.df.groupby(chat.df.index.date).agg(num_messages=('message', 'count')).sort_values(by='num_messages', ascending=False)

AttributeError: 'RangeIndex' object has no attribute 'date'

**Comment by lucasrodes** [comment]
Hi @saviols,
There was a change from v0.3.x to v0.4.0, where `date` was moved from being the index to being a column itself. You can read more at https://github.com/lucasrodes/whatstk/pull/106.

The tutorial you might be referring to (https://towardsdatascience.com/analyzing-whatsapp-chats-with-python-20d62ce7fe2d) was written for v0.3.x, therefore some instructions may not work for v0.4.x. I have updated the tutorial for the current version, so you should be able to follow it successfully. 

TL;DR: Do not use `index`

```python
chat.df.groupby(chat.df.date).agg(num_messages=('message', 'count')).sort_values(by='num_messages', ascending=False)
```

Please let me know if it works now for you.


Note: Soon there will be a new release, v0.5.0 which will be compatible with Windows out of the box. Thanks for your feedback.

**Comment by lucasrodes** [comment]
Seems fixed. Feel free to open a new issue if you run into new problems.

**Comment by lucasrodes** [comment]
Last version (v0.5.0) of the library now supports Windows: 

```
pip install whatstk
```
